c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1056,6 +1077,11 @@ class DeviceCachingAllocator {
         !block->allocated && block->event_count == 0 &&
         block->stream_uses.empty());
 
+    memory_tracker.append_alloc_free_event(
```

Comment:
some additional thoughts. IIUC, currently, we are recording allocte/free events at the allocator level instead of cudaMalloc/cudaFree level, i.e., we gonna record an allocate event even if the request is served by a cached block. If this is the case, in followup PRs, shall we add a mode to capture raw cudaMalloc/cudaFree events. This might be more helpful to debug OOM problems.

Reply:
That is a great idea. I can also add labels to allocate events, e.g., served by by a cached block, served after freeing a cached block, served after defragmentation etc?
I added this on the to-do list. Thank you